opentimestamps/core/secp256k1.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (C) 2017 The OpenTimestamps developers
+#
+# This file is part of python-opentimestamps.
+#
+# It is subject to the license terms in the LICENSE file found in the top-level
+# directory of this distribution.
+#
+# No part of python-opentimestamps including this file, may be copied,
+# modified, propagated, or distributed except according to the terms contained
+# in the LICENSE file.
+
+import hashlib
+
+from opentimestamps.core.op import UnaryOp, MsgValueError
+
+@UnaryOp._register_op
+class OpSecp256k1Commitment(UnaryOp):
+    """Map (P || commit) -> [P + sha256(P||commit)G]_x for a given secp256k1 point P
+
+    This is a unary op rather than a binary op to allow timestamps to also
+    timestamp the point itself; in the event of an ECC break this might be
+    relevant. Such a break would not affect the integrity of the commitment,
+    but knowledge of the underlying key may be interesting in its own right.
+    """
+    TAG = b'\x09'
+    TAG_NAME = 'secp256k1commitment'
+
+    def _do_op_call(self, msg):
+        if len(msg) < 33:
+            raise MsgValueError("Missing secp256k1 point")
+
+        pt = Point.decode(msg[0:33])
+
+        hasher = hashlib.sha256()
+        hasher.update(pt.encode())
+        hasher.update(msg[33:])
+        tweak = int.from_bytes(hasher.digest(), 'big')
+        tweak_pt = SECP256K1_GEN.scalar_mul(tweak)
+        final_pt = pt.add(tweak_pt)
+        return final_pt.x.to_bytes(32, 'big')
```

Comment:
With DER encoding the x-coord of the ephemeral key may be encoded in less than 32 bytes, it happens 1 time out of 512.
Something like `(final_pt.x.bit_length() + 7) // 8` instead of `32` should fix that.